its/src/test/java/com/sonar/it/scanner/msbuild/utils/BuildCommand.java

```diff
@@ -56,27 +66,44 @@ public BuildCommand addArgument(String... arguments) {
     return this;
   }
 
-  public BuildResult execute(Orchestrator orchestrator) {
-    // ToDo: SCAN4NET-312: Extract createCommand, add support for parameters, add support for MsBuild vs dotnet. Should be similar to ScannerCommand.execute()
-    // ToDo: SCAN4NET-312: Run restore & rebuild. Restore could be via NuGet (on request, or every time for simplicity?), MsBuild or dotnet
-    var environmentVariables = environment.entrySet().stream().map(x -> new EnvironmentVariable(x.getKey(), x.getValue())).toList();
-    var result = dotnetCommand == null
-      ? TestUtils.runMSBuild(orchestrator, projectDir, environmentVariables, timeout, appendArgument("/t:Restore,Rebuild"))
-      : TestUtils.runDotnetCommand(projectDir, environmentVariables, dotnetCommand, dotnetArguments());
-    assertTrue(result.isSuccess());
+  public BuildResult execute() {
+    var command = createCommand();
+    var result = new BuildResult();
+    LOG.info("Command line: {}", command.toCommandLine());
+    result.addStatus(CommandExecutor.create().execute(command, new StreamConsumer.Pipe(result.getLogsWriter()), timeout));
+    assertTrue(result.isSuccess(), "BUILD step failed.");
     return result;
   }
 
-  private String[] dotnetArguments() {
-    return dotnetCommand == "build"
-      ? appendArgument("--no-incremental")
-      : arguments.toArray(new String[0]);
+  private Command createCommand() {
+    Command command;
+    if (dotnetCommand == null) {
+      command = System.getProperty("os.name").toLowerCase().contains("windows") ?
+        Command.create(msBuildPath())
+        // Using the msbuild command from the dotnet CLI allows to use the same parameters as the Windows version
+        // `dotnet build` is a wrapper around `msbuild`
```

Comment:
I find the current version bit confusing.
```suggestion
        // Using the msbuild command from the dotnet CLI allows to use the same parameters as MsBuild.exe on Windows
```